### DISCUSSION TITLE
How and when to use `LOCAL_NETWORK_ALLOWLIST`?

### DISCUSSION THREAD

**Discussion by jchonig** [gogs/gogs#7048] [Q&A] [answered]
This is briefly mentioned in the release nodes, but you can expand on it's use?

I run gogs in a docker container and have a couple repos that send webhooks.

Thanks.

Jeff

**Comment by unknwon** [comment, marked as answer]
When you want to migrate repositories or send webhooks from/to a local network address (not a public IP address on internet). If everything is working from your perspective, there is no need to take action.

To use it:

```ini
[security]
# specific network address, separated by commas, no port needed
LOCAL_NETWORK_ALLOWLIST = bitbucket.local, sourcegraph.test
```

**Reply by tjh19971228** [reply]
how to use in docker-compose

**Reply by johnnestebann** [reply]
@unknwon same question

**Reply by liukonen** [reply]
I have 2 services on the same machine (Gogs -> webhook - > Jenkins build pipeline)... when I go into the gogs/config/app.ini and add the above line, opening my internal IP as a local network, I still get the same issue

**Reply by wandersonwhcr** [reply]
This is not working on Kubernetes. I have Gogs as service on `gogs.gogs` and Drone on `drone.drone`. Webhook is defined as `http://drone.drone/hook`. I added to `LOCAL_NETWORK_ALLOWLIST` value `drone.drone` and all requests are blocked. Also, I added `drone.drone.svc.cluster.local` to this variable and I received the same error.

I rolled back to `0.12.8` and it worked.

**Reply by wandersonwhcr** [reply]
Sorry! Gogs `0.12.7` :pray: 

https://github.com/wandersonwhcr/k8s-examples/blob/main/gogs/statefulset.yaml

**Reply by unknwon** [reply]
NVM there is another bug that was fixed but not yet released: https://github.com/gogs/gogs/issues/7074 😅  so 0.12.7 is still a good version to roll back to.

**Reply by wandersonwhcr** [reply]
TY @unknwon

BTW your project is awesome <3

**Reply by unknwon** [reply]
FYI https://github.com/gogs/gogs/pull/7111, I don't know why I did not think of such easy escape hatch initially 🤷‍♂️

**Reply by unknwon** [reply]
0.12.10 is currently planned to be released on July 30th (not a commitment 🤫).

**Reply by kankan-k2** [reply]
I have a similar requirement where I want localhost to used for a webhook. I am running gogs inside a docker container. How do I use this inside a docker container?
I created a custom/config/app.ini file inside the /app/gogs/ folder and put just this:
/app/gogs # cat custom/conf/app.ini 
[security]
\# specific network address, separated by commas, no port needed
LOCAL_NETWORK_ALLOWLIST = *

But it doesn't work. 

I am using the latest label of the gogs docker image

**Reply by kankan-k2** [reply]
actually solved it.
based on logs in the docker, custom config is picked up from /data/gogs/conf/app.ini. Added this option/parameter there and restarted the container. gogs seemed to have picked up the change and allowed me to create the webhook using localhost.

**Reply by kankan-k2** [reply]
one problem which I found while trying to use localhost (or 127.0.0.1 or 0.0.0.0). gogs is unable to send/make requests to localhosts. here is a sample response:
Delivery: Post "http://localhost:8890/monitor": dial tcp [::1]:8890: connect: connection refused

But if I replace the localhost with the actual IP of the localhost/my system (the local home network IP or if connected to VPN, then the IP provided by that VPN service), then gogs is able to make the HTTP requests.

Not sure if this is a bug or an intended behavior.

**Reply by unknwon** [reply]
> Delivery: Post "http://localhost:8890/monitor": dial tcp [::1]:8890: connect: connection refused

Judging based on the error message, I think it is an intended behavior.
